src/transformers/trainer.py

```diff
@@ -5152,6 +5153,9 @@ def get_batch_samples(self, epoch_iterator, num_batches):
             except (TypeError, AttributeError):
                 pass
 
+        if num_items_in_batch is None:
+            num_items_in_batch = torch.tensor(0).to(self.args.device)
+
```

Comment:
Let's do that only inside `average_tokens_across_devices` as `num_items_in_batch` should be an int. We need a tensor only for the gather

Reply:
Then I assume it is okay for this function to return `None` for num_items_in_batch ?

Reply:
Yeah that's fine

Reply:
Thanks! made the change